src/pkg/shub/client/util_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/singularityware/singularity/src/pkg/test"
+)
+
+var (
+	validShubURIs = []string{
+		`shub://username/container`,
+		`shub://username/container:tag`,
+		`shub://username/container@00000000000000000000000000000000`,
+		`shub://registry/username/container`,
+		`shub://registry/with/levels/username/container`,
+		`shub://registry/user-name/container-with-dash`,
+		`shub://registry/username/container.with.period`,
+		`shub://username/container:tag-with-dash`,
+		`shub://username/container:tag_wtih_underscore`,
+		`shub://username/container:tag.with.period`,
+		`shub://myprivateregistry.sylabs.io/sylabs/container:latest`,
+	}
+)
+
+// TestShubParser checks if the Shub ref parser is working as expected
+func TestIsShubPullRef(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	invalidShubURIs := []string{
+		`shub://username/`,
+		`shub://username/container:`,
+		`shub://username/container@`,
+		`shub://username/container@0000000000000000000000000000000`,
+		`shub://username/container@000000000000000000000000000000000`,
+		`shub://username/container@abcdefghijklmnopqrstuvwxyz123456`,
```

Comment:
This one would be a valid `@<branch>`

Reply:
is not valid due you need a `:` before the `@`

Reply:
gotcha!